### PR TITLE
gracefully handle caching and user-stopping downloads

### PR DESCRIPTION
**Pull request by lawinslow**
Thoughts on this?

### CODE REVIEW

File: R/lsat_image.R [hunk 1]
```diff
@@ -30,12 +30,17 @@ lsatGET <- function(url, dat, overwrite, ...) {
     message("File in cache")
     return(fpath)
   } else {
-    dir.create(dirname(fpath), showWarnings = FALSE, recursive = TRUE)
-    res <- GET(url, write_disk(path = fpath, overwrite = overwrite), ...)
-    # delete file if error, and stop message
+    temp_path = tempfile()
```
Comment (sckott) [inline R/lsat_image.R:33]: if we went with `tempfile()` I think we'd need to make sure to close that connection on exit

**Comment by sckott** [comment]
I don't know, but I wonder what is better: 
- write a file once to tempfile, then move it to another location on success - as you've done
- or, write to final location, and on failure, `unlink(path)` to delete the partial file

File: R/lsat_image.R (hunk 1 above)
Comment (lawinslow) [inline R/lsat_image.R:33]: Hmmm, do we? How?

**Comment by lawinslow** [comment]
I don't think you can use your second option. If the user interrupts execution, will the `unlink` ever be reached?

**Comment by lawinslow** [comment]
Fixed the overwrite use...

File: R/lsat_image.R (hunk 1 above)
Comment (sckott) [inline R/lsat_image.R:33]: I don't know for sure, maybe not, forgot about it for now

**Comment by sckott** [comment]
You're probably right, forget about unlink

Did you check that the tempfile created is gone after an error? I assume it is

**Comment by lawinslow** [comment]
My Kiwi internet is slow, so I haven't tested it in all circumstances. But I think it's handled correctly.

**Pull request merged**